### ISSUE TITLE
Performance: checking the PhoneNumber type in the cellForRowAtIndexPath

### ISSUE THREAD

**Issue by rsaarloos** [marmelroy/PhoneNumberKit#75]
Hi,

In my app I load the contact list of the user. In this list I want to make non-mobile numbers greyed-out and non-selectable. To do this in the cellForRowAtIndexPath I check the type of the phone number. As I have read and seen in the code this computed value is quite heavy (as it involves quite a lot of regular expressions). This impact can be seen when scrolling through the list.

Is there a way to improve the performance in this scenario? For example is it possible to just check if the number is of .Mobile type, without also determining if it also is of other types (what is currently happening)?

Thanks!

**Comment by marmelroy** [comment]
Hi @rsaarloos, 

Thank you for your question. 

Are you using the newest version of PhoneNumberKit?  As of version 1.0, the type is no longer a computer property and it is determined on parse. This should be fast enough for most use cases (especially if you are conservative about the number of parses you do, I recommend using the special function that parses an array of string phone numbers). 

There currently isn't a way to check just for a Mobile type but this is something I could consider adding...

**Comment by rsaarloos** [comment]
Thanks for the quick reply.

Not yet upgraded to 1.x due to my project not being Swift 3 ready. I planned to do this in the coming weeks. Will keep you up to date if this significantly improves performance.

**Comment by rsaarloos** [comment]
I upgraded to Swift 3.0 and the latest version of the PhoneNumberKit and the issue is now less apparent.

I tried out the special function for parsing an array of phone numbers and was quite disappointed with the performance. On a dataset of 2000 phone numbers it took 3 seconds to read them of the phone's contacts and about 20 seconds to parse them with PhoneNumberKit. In the resulting array I also lost 600 phone numbers as the parsing function only returned 1400 results.

This test was done on an iPhone 5. Maybe I'm doing something wrong here as the main page talks about 1000 parses -> ~0.4 seconds.

Any ideas?
